plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2936,17 +2939,25 @@ public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession s
         String dependsOnTables = currentSnapshot
                 .map(snapshot -> snapshot.summary().getOrDefault(DEPENDS_ON_TABLES, ""))
                 .orElse("");
+        boolean dependsOnTableFunctions = currentSnapshot
+                .map(snapshot -> Boolean.valueOf(snapshot.summary().getOrDefault(DEPENDS_ON_FUNCTIONS, "false")))
+                .orElse(false);
+
+        Optional<Instant> refreshTime = currentSnapshot.map(snapshot -> snapshot.summary().get(TRINO_QUERY_START_TIME))
+                .map(Instant::parse)
+                .or(() -> currentSnapshot.map(snapshot -> Instant.ofEpochMilli(snapshot.timestampMillis())));
+
+        if (dependsOnTableFunctions) {
+            // we will never know if table function returns stale value or not
```

Comment:
```suggestion
            // It can't be determined whether table function returns stale value or not
```

Reply:
TBH neither of these sentences are correct :D